documentation/glossary.md

```diff
@@ -0,0 +1,151 @@
+
+Filecoin Pin reuses existing Data Sets by default, matching on [metadata keys](#metadata-keys) (`source='filecoin-pin'`). If multiple exist, it uses the one storing the most data.
+
+## FIL
+
+FIL is Filecoin's native token.  While [Filecoin Onchain Cloud](#filecoin-onchain-cloud) storage is denominated in [USDFC](#usdfc), transactions on the Filecoin blockchain (e.g., adding a [Piece](#piece)) need to be paid for using FIL.
+
+## Filecoin Pay
+
+Filecoin Pay is a generic payment solution between users and various [Filecoin Onchain Cloud](#filecoin-onchain-cloud) services.
+
+Learn more at https://github.com/FilOzone/filecoin-pay
+
+## Filecoin Onchain Cloud
+
+https://filecoin.cloud.
+
+This is the collection of DePIN services offered on Filecoin using shared/consistent payment infrastructure in [Filecoin Pay](#filecoin-pay).  [Filecoin Warm Storage Service](#filecoin-warm-storage-service) is the initial service offering.
+
+This is often abbreviated as “FOC”, which yes, does phonetically resonate with more colorful language 😉.
+
+## Filecoin Pin
+
+https://github.com/FilOzone/filecoin-pin
+
+Serves as an IPFS-oriented set of tools for interfacing with [Filecoin Onchain Cloud](#filecoin-onchain-cloud) built on top of [Synapse](#synapse).
+
+## `filecoin-pin`
+
+`filecoin-pin` is a CLI tool affordance for [Filecoin Pin](#filecoin-pin).
+
+## filecoin-pin-website
+
+https://github.com/filecoin-project/filecoin-pin-website
+
+Example of [Filecoin Pin](#filecoin-pin) in action within a web-browser.  Its purposes are:
+
+1. Demonstrate that Filecoin Pin is usable.  Drag and drop and you're good to go!
+2. Serve as a starter or inspiration for dApp builders wanting to use [Filecoin Onchain Cloud](#filecoin-onchain-cloud).
+
+filecoin-pin-website is also hosted at [pin.filecoin.cloud](http://pin.filecoin.cloud), with hardcoded wallet and [session key](#session-key) on the [Calibration](#calibration-network) network.  In future, [integration with tools like Metamask will be supported](https://github.com/filecoin-project/filecoin-pin-website/issues/77).  
+
+## Filecoin Warm Storage Service
+
+This is the entry point smart contract for using the warm storage functionality offered in [Filecoin Onchain Cloud](#filecoin-onchain-cloud).  It acts as a validator for payment settlements, ensuring the warm storage service is actually delivered before payments are released to the [Storage Provider](#storage-provider).
+
+## IPFS Root CID
+
+The CID for the root of a merkle DAG that is usually encoding a file or directory as UnixFS.  Since each `filecoin-pin add` creates a [CAR](#car), regardless if passed a file or directory, there is a single root corresponding to root of the Merkle DAG made out of encoding the file or directory as UnixFS.
```

Comment:
```suggestion
The CID for the root of a merkle DAG that is usually encoding a file or directory as UnixFS.  Since each `filecoin-pin add` creates a [CAR](#car), regardless if passed a file or directory, there is a single root corresponding to root of the Merkle DAG made out of encoding the file or directory as UnixFS. Typically this will be presented in base32, beginning with `bafy` and be 59 characters long.
```